mikrotik/resource_dhcp_lease.go

```diff
@@ -0,0 +1,139 @@
+package mikrotik
+
+import (
+	"fmt"
+
+	"github.com/ddelnano/terraform-provider-mikrotik/client"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceLease() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceLeaseCreate,
+		Read:   resourceLeaseRead,
+		Update: resourceLeaseUpdate,
+		Delete: resourceLeaseDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"address": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"macaddress": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+		},
+	}
+}
+
+func resourceLeaseCreate(d *schema.ResourceData, m interface{}) error {
+	address := d.Get("address").(string)
+	macaddress := d.Get("macaddress").(string)
+
+	c := m.(client.Mikrotik)
+
+	r, err := c.AddDhcpLease(address, macaddress)
+	if err != nil {
+		return err
+	}
+
+	// If API is successful we should only get a single sentence and list back like so
+	// 2019/02/28 20:13:15 !done @ [{`ret` `*14`}]
+	var id string
+	for _, reply := range r.Re {
+		for _, item := range reply.List {
+			if item.Key == ".id" {
+				id = item.Value
+			}
+		}
+	}
+
+	lease := &client.DhcpLease{
+		Id:         id,
+		Address:    address,
+		MacAddress: macaddress,
+	}
+	leaseToData(lease, d)
+	return nil
+}
+
+func resourceLeaseRead(d *schema.ResourceData, m interface{}) error {
+	c := m.(client.Mikrotik)
+
+	lease, err := c.FindDhcpLease(d.Id())
+
+	// TODO: Ignoring this error can cause all resources to think they
+	// need to be created. We should more appropriately handle this. The
+	// error where the DHCP lease is not found is not actually an error and
+	// needs to be disambiguated from real failures
+	if err != nil {
+		d.SetId("")
+		return nil
+	}
+
+	// FIXME
```

Comment:
This doesn't seem to be copy and pasted so what is the context on this?

Reply:
Removed